product_supplier_intercompany/models/pricelist.py

```diff
@@ -75,10 +68,14 @@ def _add_product_to_synchronize(self, todo):
                 todo[pricelist]["products"] |= record.product_id
             elif record.product_tmpl_id:
                 todo[pricelist]["templates"] |= record.product_tmpl_id
-            else:
+            # there is another item type
+            elif len(todo[pricelist].items()) > 2:
```

Comment:
I do not understand your change here.
The initial aim was to raise an error we we have item price not applied on product or template

Reply:
That was a rushed way to deal with the default pricelists. The motivation was to raise an error if there was anything else than products or templates in todo, and in any other case just don't deal with it.

When you create a pricelist, it creates a default pricelist item. That default pricelist item has no product_id or product_templ_id. Thus creating an intercompany pricelist raises an error.

Do you think we can ignore other alternatives for now and, in case pricelist has no product_tmpl_id or product_id, pass it silently ?

Reply:
I think another option to solve this is to do a check when we set a pricelist from non-intercompany to intercompany, do a check on all pricelist items so that they have a product_id or product_tmpl_id and raise a specific error 'Intercompany pricelist lines must have either a product or a product variant assigned to them'

Reply:
I went for the above solution https://github.com/akretion/ak-odoo-incubator/commit/28164760291256ec682783373062fe52d821dc55